report_dynamic/models/report_dynamic.py

```diff
@@ -0,0 +1,271 @@
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class ReportDynamic(models.Model):
+    _name = "report.dynamic"
+    _description = "Dynamically create reports"
+
+    name = fields.Char(required=True)
+    model_id = fields.Many2one("ir.model")
+    # Inform the user about configured model_id
+    # in template
+    model_name = fields.Char(related="model_id.name")
+    model_model = fields.Char(related="model_id.model", string="Tech name of model_id")
+    res_id = fields.Integer()
+    resource_ref = fields.Reference(
+        string="Target record",
+        selection="_selection_target_model",
+        compute="_compute_resource_ref",
+        inverse="_inverse_resource_ref",
+        store=True,
+    )
+    wrapper_report_id = fields.Many2one("ir.ui.view", domain="[('type', '=', 'qweb')]")
+    template_id = fields.Many2one(
+        "report.dynamic",
```

Comment:
```suggestion
        comodel_name="report.dynamic",
```
for the sake of consistency I'd apply the same to all fields (all params with their names)